engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java

```diff
@@ -875,7 +875,8 @@ public static String getSocketConnectString(InetSocketAddress socketAddress)
   public static void addAttributeToArgs(Attribute<String> attribute, Context context, List<CharSequence> vargs)
   {
     if (context.getValue(attribute) != null) {
-      vargs.add(String.format("-D%s=$'%s'", attribute.getLongName(), context.getValue(attribute).replaceAll("['\"$\\\\]", "\\\\$0")));
+      vargs.add(String.format("-D%s=$'%s'", attribute.getLongName(), context.getValue(attribute)
```

Comment:
It is necessary to verify that the launched container has properties properly set. Please see StramMiniClusterTest.testSetupShutdown(). It launches a separate container (java process) using `vargs`.